lib/logstash/codecs/line.rb

```diff
@@ -30,28 +31,30 @@ def register
     @converter.logger = @logger
   end
 
-  public
   def decode(data)
     @buffer.extract(data).each do |line|
-      yield LogStash::Event.new("message" => @converter.convert(line))
+      yield eventify(line)
     end
   end # def decode
 
-  public
   def flush(&block)
     remainder = @buffer.flush
     if !remainder.empty?
-      block.call(LogStash::Event.new("message" => @converter.convert(remainder)))
+      block.call(eventify(remainder))
     end
   end
 
-  public
   def encode(event)
-    if event.is_a? LogStash::Event and @format
+    if !@format.nil? && event.is_a?(LogStash::Event)
```

Comment:
It is to do short-circuit evaluation as `@format` is crucial to the branch logic, if LHS is false, ruby will not do the RHS.